dig/dig.go

```diff
@@ -211,25 +211,27 @@ func (g *Graph) registerConstructor(c interface{}) error {
 // When a new constructor is being inserted, detect any present cycles
 func (g *Graph) detectCycles(n *funcNode) error {
 	visited := make(map[string]bool)
-	return g.recursiveDetectCycles(n, visited)
+	l := list.New()
+	return g.recursiveDetectCycles(n, visited, l)
 }
 
 // DFS and tracking if same node is visited twice
-func (g *Graph) recursiveDetectCycles(n graphNode, visited map[string]bool) error {
-	if visited[n.id()] {
-		return errCycle
-	}
-
+func (g *Graph) recursiveDetectCycles(n graphNode, visited map[string]bool, l *list.List) error {
 	visited[n.id()] = true
+	l.PushBack(n.id())
```

Comment:
You need to remove a Type from the list(with defer?), otherwise, you'll include a branch info, that is not a cycle, e.g.:
```
T1
|  \
T2  T3
      \
       T1
```
Will print T1 -> T2 -> T3 -> T1, but I'd expect only relevant info: T1 -> T3 -> T1.

Easier approach would be to return (wrap) type in the error and combine them with the current type it if error is not nil.